cjsmin/src/chess.ts

```diff
@@ -1787,7 +1787,7 @@ export class Chess {
         };
         result.push({
           move: prettyMove,
-          board: this._board,
+          board: [...this._board], // need to copy the board, not pass a reference to it since we are aggregating and doing one game at a time. Copying this is SLOW
```

Comment:
ah yes - and this is causing issues with how threefold repetition check is working

Reply:
slow, but necessary? any way around it?

Reply:
Yes, if we did analysis at the move level instead of game level and used sliding windows where required. Or modified board state in cases where we did need to track it over time, rather than having a copy for every move